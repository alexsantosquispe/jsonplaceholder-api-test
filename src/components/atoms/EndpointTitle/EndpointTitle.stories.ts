import type { Meta, StoryObj } from '@storybook/react';

import { EndpointTitle } from './EndpointTitle';

const meta = {
  title: 'Atoms/EndpointTitle',
  component: EndpointTitle
} satisfies Meta<typeof EndpointTitle>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    methodLabel: 'GET',
    label: 'Posts'
  }
};
