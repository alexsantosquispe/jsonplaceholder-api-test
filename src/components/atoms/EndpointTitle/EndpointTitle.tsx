interface EndpointTitleProps {
  methodLabel: string;
  label: string;
}

export const EndpointTitle = ({ methodLabel, label }: EndpointTitleProps) => {
  return (
    <div className="flex gap-4">
      <span className="rounded-sm bg-green-100 px-2 py-1 text-base font-normal text-green-800 uppercase">
        {methodLabel}
      </span>
      <h1 className="text-2xl font-bold">{label}</h1>
    </div>
  );
};
