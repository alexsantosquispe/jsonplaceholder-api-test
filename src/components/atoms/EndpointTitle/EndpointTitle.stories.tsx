import type { Meta, StoryObj } from '@storybook/react';

import { EndpointTitle } from './EndpointTitle';
import { Wrapper } from '../../../utils/stories-utils';

const meta = {
  title: 'Atoms/EndpointTitle',
  component: EndpointTitle,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof EndpointTitle>;

export default meta;

type Story = StoryObj<typeof EndpointTitle>;

export const Default: Story = () => {
  return (
    <Wrapper>
      <EndpointTitle methodLabel="GET" label="Posts" />
    </Wrapper>
  );
};

Default.args = {};
