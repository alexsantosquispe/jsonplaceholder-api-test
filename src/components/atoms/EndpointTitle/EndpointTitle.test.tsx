import { render, screen } from '@testing-library/react';

import { EndpointTitle } from './EndpointTitle';

describe('EndpointTitle', () => {
  describe('Styles', () => {
    it('should render correctly', () => {
      render(<EndpointTitle methodLabel="GET" label="Posts" />);

      expect(screen.getByText('GET')).toBeInTheDocument();
      expect(screen.getByText('Posts')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
      const { container } = render(
        <EndpointTitle methodLabel="GET" label="Posts" />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
