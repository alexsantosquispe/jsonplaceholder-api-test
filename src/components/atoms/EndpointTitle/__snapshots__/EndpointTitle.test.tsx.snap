// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndpointTitle Styles should match snapshot 1`] = `
<div>
  <div
    class="flex gap-4"
  >
    <span
      class="rounded-sm bg-green-100 px-2 py-1 text-base font-normal text-green-800 uppercase dark:bg-green-950 dark:text-green-400"
    >
      GET
    </span>
    <h1
      class="text-2xl font-bold dark:text-white"
    >
      Posts
    </h1>
  </div>
</div>
`;
